src/form/wi.jl

```diff
@@ -0,0 +1,139 @@
+export
+    SOCWIPowerModel, SOCWIForm
+""
+abstract type AbstractWIForm <: AbstractPowerFormulation end
+
+""
+abstract type SOCWIForm <: AbstractWIForm end
+
+""
+const SOCWIPowerModel = GenericPowerModel{SOCWIForm}
+
+"default SOC constructor"
+SOCWIPowerModel(data::Dict{String,Any}; kwargs...) = GenericPowerModel(data, SOCWIForm; kwargs...)
+
+""
+function variable_current(pm::GenericPowerModel{T}, n::Int=pm.cnw; kwargs...) where T <: AbstractWIForm
+    variable_series_current_magnitude_sqr(pm, n; kwargs...)
+end
+
+""
+function variable_branch_flow(pm::GenericPowerModel{T}, n::Int=pm.cnw; kwargs...) where T <: AbstractWIForm
+    variable_active_branch_flow(pm, n; kwargs...)
+    variable_reactive_branch_flow(pm, n; kwargs...)
+    variable_current(pm, n; kwargs...)
+end
+
+""
+function variable_voltage(pm::GenericPowerModel{T}, n::Int=pm.cnw; kwargs...) where T <: AbstractWIForm
+    variable_voltage_magnitude_sqr(pm, n; kwargs...)
+end
+
+"do nothing, this model does not have complex voltage constraints"
+function constraint_voltage(pm::GenericPowerModel{T}, n::Int) where T <: AbstractWIForm
+end
+
+
+"""
+Creates Ohms constraints (yt post fix indicates that Y and T values are in rectangular form)
+"""
+function constraint_ohms_yt_from(pm::GenericPowerModel{T}, n::Int, f_bus, t_bus, f_idx, t_idx, g, b, c, tr, ti, tm) where T <: AbstractWIForm
```

Comment:
I agree this does not make a lot of sense.  I think this formulation needs a new OPF formulation that better reflect its construction.

Reply:
ok, I agree, made a proposal that still allows you to use BIM formulations in the BFM problem type and vice-versa.